### PR TITLE
adding a --cached flag for yarn start and build cache for PR/CI pipeline

### PR DESCRIPTION
**Pull request by kenotron**
#### Pull request checklist

- [x] Include a change request file using `$ yarn change`

#### Description of changes

Okay, so we can take advantage of some caching to speed up several things:

1. innerloop could start super fast if we had hard-source-webpack-plugin - however, I'm not really sure what it feels like across branch switching, so would like to see if we can have this in behind an arg to test it out

2. build cache in pipeline is in beta, and we definitely can use this to speed up our builds

#### Focus areas to test

(optional)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/10229)

### CODE REVIEW

File: scripts/webpack/webpack-resources.js [hunk 1]
```diff
@@ -139,81 +140,86 @@ module.exports = {
...
-    const config = merge(
-      {
-        devServer: {
-          inline: true,
-          port: 4322,
-          contentBase: path.resolve(process.cwd(), 'dist')
-        },
+    return env => {
+      env = env || {};
+      console.log('SHOULD CACHE: ', Boolean(!process.env.TF_BUILD && env.cached));
```
Comment (jdhuntington) [inline scripts/webpack/webpack-resources.js]: Intentional?

File: scripts/tasks/webpack.js [hunk 2]
```diff
@@ -14,8 +14,9 @@ exports.webpackDevServer = async function() {
   if (fs.existsSync(configPath)) {
     const webpackDevServerPath = require.resolve('webpack-dev-server/bin/webpack-dev-server.js');
     const execSync = require('../exec-sync');
-
-    execSync(`node ${webpackDevServerPath} --config ${configPath} --port ${port} --open`);
+    const cmd = `node ${webpackDevServerPath} --config ${configPath} --port ${port} --open ${argv().cached ? '--env.cached=true' : ''}`;
+    console.log('RUNNING: ', cmd);
```
Comment (jdhuntington) [inline scripts/tasks/webpack.js]: Intentional?

Comment (kenotron) [reply scripts/tasks/webpack.js]: Not that I like this, but accidentally left this. Will convert to logger.info when the first iteration run is done to compare the yarn install speeds after build cache has been established.

File: azure-pipelines.yml [hunk 3]
```diff
@@ -16,9 +18,15 @@ steps:
...
+  - task: CacheBeta@0
+    inputs:
+      key: yarn | $(Agent.OS) | yarn.lock
+      path: $(YARN_CACHE_FOLDER)
+    displayName: Cache Yarn packages
+
   - script: |
       yarn
-    displayName: yarn
+    displayName: yarn --frozen-lockfile
```
Comment (ecraig12345) [inline azure-pipelines.yml]: Should you do this in the actual command too?

File: package.json [hunk 4]
```diff
@@ -15,8 +15,8 @@
     "clean-all": "node ./scripts/clean-all.js",
     "vrtest": "cd apps && cd vr-tests && npm run screener",
     "a11ytest": "cd apps && cd a11y-tests && npm run test",
-    "start": "cd apps && cd fabric-website-resources && npm start",
-    "start-exp": "cd packages && cd experiments && npm start",
+    "start": "yarn workspace @uifabric/fabric-website-resources start",
```
Comment (ecraig12345) [inline package.json]: What does this do? Should we be doing it for other commands too?

File: azure-pipelines.yml (hunk 3 above)
Comment (kenotron) [reply azure-pipelines.yml]: okay !!

File: package.json (hunk 4 above)
Comment (kenotron) [reply package.json]: we should - we couldn't pass any args if we used the && syntax.

File: scripts/webpack/webpack-resources.js (hunk 1 above)
Comment (kenotron) [reply scripts/webpack/webpack-resources.js]: nope!

File: scripts/tasks/webpack.js (hunk 2 above)
Comment (kenotron) [reply scripts/tasks/webpack.js]: yarn build cache was slower!!

**Pull request merged**